frontend/src/modules/editor/components/Editor.js

```diff
@@ -78,6 +96,11 @@ export class EditorBase extends React.Component<InternalProps> {
     }
 
     sendTranslation = (ignoreWarnings: ?boolean) => {
+        if (this.state.isSendTranslationDisabled) {
+            return;
+        }
+        this.setState({ isSendTranslationDisabled: true });
```

Comment:
These could be refactored into a Higher-Order component, allowing to prevent to pattern repetition. See https://reactjs.org/docs/higher-order-components.html

Reply:
I gave HOC a [try](https://github.com/mozilla/pontoon/pull/1273/commits/c1c0ecd4705bae4bf8b53d4ab92b749031d9aeb5), and now I'm not sure it's the right tool for the job.

My understanding is that we're passing props from the WrappedComponent to HOC, which then adds additional props, and does something with them. We don't seem to have access to those HOC-specific props in the WrappedComponent.

In our use case here we'd need access, though. E.g. in `render()` and when restoring actions.